src/Mappers/TransactionMapper.php

```diff
@@ -263,18 +261,19 @@ private static function getField(BaseTransaction $transaction, \DOMElement $elem
         return $fieldElement->textContent;
     }
 
-    private static function getFieldAsMoneyObject(
+    private static function getFieldAsMoney(
         BaseTransaction $transaction,
         \DOMElement $element,
-        string $fieldTagName
+        string $fieldTagName,
+        Currency $currency
     ): ?Money {
         $fieldValue = self::getField($transaction, $element, $fieldTagName);
 
         if ($fieldValue === null) {
             return null;
         }
 
-        return Money::EUR((string)(100 * $fieldValue));
+        return new Money((string)(100 * $fieldValue), $currency);
```

Comment:
hmm this `100 *` is going to cause some issues, guess we should use `DecimalParser` but good enough for now for me.